packages/opentelemetry-resources/src/platform/node/detectors/AwsBeanstalkDetector.ts

```diff
@@ -33,7 +34,8 @@ class AwsBeanstalkDetector implements Detector {
 
   async detect(config: ResourceDetectionConfigWithLogger): Promise<Resource> {
     try {
-      const rawData = await this._getData();
+      const readFile = util.promisify(fs.readFile);
```

Comment:
Move this outside of the class/method. It is module-level

Reply:
Make sense, let me do it

Reply:
It seems that when promisify the ```fs.readFile``` outside of the class, ```sinon.sandbox.stub``` cannot simply stub readFile function by ```sandbox.stub(fs, 'readFile').yields(null, JSON.stringify(data));``` anymore. 
I searched around and did not find any useful info, I currently tend to add ```export const readFile = util.promisify(fs.readFile);``` in main function and stub it in unit test. Is that good? Any suggestions for that?

Reply:
I have actually blocked here for couple of hours, I am pretty confused why when ```readFile = util.promisify(fs.readFile);``` is inside the method, ```sandbox.stub(fs, 'readFile')``` works and when we take it out of the class ```sandbox.stub(fs, 'readFile')``` failed.